plugins/inputs/cisco_telemetry_mdt/cisco_telemetry_mdt.go

```diff
@@ -50,13 +52,976 @@ type CiscoTelemetryMDT struct {
+    c.nxpathMap[key]["register"] = "string"
+    c.nxpathMap[key]["sgexpire"] = "string"
+    c.nxpathMap[key]["oif-bf-str"] = "string"
+    c.nxpathMap[key]["mcast-addrs"] = "string"
+    c.nxpathMap[key]["rp-addr"] = "string"
+    c.nxpathMap[key]["immediate-bf-str"] = "string"
+    c.nxpathMap[key]["sgr-prune-list-bf-str"] = "string"
+    c.nxpathMap[key]["context-name"] = "string"
+    c.nxpathMap[key]["intf-name"] = "string"
+    c.nxpathMap[key]["immediate-timeout-bf-str"] = "string"
+    c.nxpathMap[key]["rp-local"] = "string"
+    c.nxpathMap[key]["sgrexpire"] = "string"
+    c.nxpathMap[key]["timeout-bf-str"] = "string"
+    c.nxpathMap[key]["timeleft"] = "string"
+
+    return nil
+}
+
+func (c *CiscoTelemetryMDT) InitPimRp() error {
+    key := "show ip pim rp vrf all"
+    c.nxpathMap[key] = make(map[string]string, 20)
+    c.nxpathMap[key]["is-bsr-forward-only"] = "string"
+    c.nxpathMap[key]["is-rpaddr-local"] = "string"
+    c.nxpathMap[key]["bsr-expires"] = "string"
+    c.nxpathMap[key]["autorp-expire-time"] = "string"
+    c.nxpathMap[key]["rp-announce-policy-name"] = "string"
+    c.nxpathMap[key]["rp-cand-policy-name"] = "string"
+    c.nxpathMap[key]["is-autorp-forward-only"] = "string"
+    c.nxpathMap[key]["rp-uptime"] = "string"
+    c.nxpathMap[key]["rp-owner-flags"] = "string"
+    c.nxpathMap[key]["df-bits-recovered"] = "string"
+    c.nxpathMap[key]["bs-timer"] = "string"
+    c.nxpathMap[key]["rp-discovery-policy-name"] = "string"
+    c.nxpathMap[key]["arp-rp-addr"] = "string"
+    c.nxpathMap[key]["auto-rp-addr"] = "string"
+    c.nxpathMap[key]["autorp-expires"] = "string"
+    c.nxpathMap[key]["is-autorp-enabled"] = "string"
+    c.nxpathMap[key]["is-bsr-local"] = "string"
+    c.nxpathMap[key]["is-autorp-listen-only"] = "string"
+    c.nxpathMap[key]["autorp-dis-timer"] = "string"
+    c.nxpathMap[key]["bsr-rp-expires"] = "string"
+    c.nxpathMap[key]["static-rp-group-map"] = "string"
+    c.nxpathMap[key]["rp-source"] = "string"
+    c.nxpathMap[key]["autorp-cand-address"] = "string"
+    c.nxpathMap[key]["autorp-up-time"] = "string"
+    c.nxpathMap[key]["is-bsr-enabled"] = "string"
+    c.nxpathMap[key]["bsr-uptime"] = "string"
+    c.nxpathMap[key]["is-bsr-listen-only"] = "string"
+    c.nxpathMap[key]["rpf-nbr-address"] = "string"
+    c.nxpathMap[key]["is-rp-local"] = "string"
+    c.nxpathMap[key]["is-autorp-local"] = "string"
+    c.nxpathMap[key]["bsr-policy-name"] = "string"
+    c.nxpathMap[key]["grange-grp"] = "string"
+    c.nxpathMap[key]["rp-addr"] = "string"
+    c.nxpathMap[key]["anycast-rp-addr"] = "string"
+
+    return nil
+}
+
+func (c *CiscoTelemetryMDT) InitPimStats() error {
+    key := "show ip pim statistics vrf all"
+    c.nxpathMap[key] = make(map[string]string, 20)
+    c.nxpathMap[key]["vrf-name"] = "string"
+
+    return nil
+}
+
+func (c *CiscoTelemetryMDT) InitIntfBrief() error {
+    key := "show interface brief"
+    c.nxpathMap[key] = make(map[string]string, 20)
```

Comment:
is there any reason to give all of these maps a length of 20?

Reply:
we just give it a little bit extra space instead of calculate the exact number of keys of each command, in latest commit we fixed everything with number of entries, in few scenario we just use 20 as approximation.

Reply:
Well the maps are dynamic and can be initialized with zero. Giving a capacity only helps to reduce the reallocations in case you have an idea where you end-up.